com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -244,12 +238,23 @@ private boolean shouldUpdateTags(CloudServiceExtended service, List<String> exis
         if (service.isUserProvided()) {
             return false;
         }
-        existingServiceTags = ObjectUtils.defaultIfNull(existingServiceTags, Collections.<String> emptyList());
+        existingServiceTags = ObjectUtils.defaultIfNull(existingServiceTags, Collections.<String>emptyList());
         return !existingServiceTags.equals(service.getTags());
     }
 
-    private boolean shouldUpdateCredentials(CloudServiceExtended service, Map<String, Object> credentials) {
-        return !Objects.equals(service.getCredentials(), credentials);
+    private boolean shouldUpdateCredentials(CloudServiceExtended service, CloudService existingService, CloudControllerClient client) {
+        if (existingService == null) {
+            return MapUtils.isEmpty(service.getCredentials());
+        }
+        try {
+            Map<String, Object> serviceParameters = client.getServiceParameters(existingService.getMetadata().getGuid());
+            getStepLogger().debug("Existing service parameters: " + secureSerializer.toJson(serviceParameters));
+            return !Objects.equals(service.getCredentials(), serviceParameters);
```

Comment:
Why are we checking for equality between credentials and parameters?
Shouldn't we make the corresponding changes to our cloud model?

Reply:
will be done in a separate pull req.